### ISSUE TITLE
Named palettes yield unexpected result

### ISSUE THREAD

**Issue by banesullivan** [girder/large_image#855]
I would assume the following two styles to be equivalent

<table>
<tr>
<th>
A
</th>
<th>
B
</th>
</tr>

<tr>

<td>
<pre>
 {
  bands: [
    {band: 1, palette: ['#000', '#f00']},  // red
    {band: 2, palette: ['#000', '#0f0']},  // green
    {band: 3, palette: ['#000', '#00f']}   // blue
  ]
}
</pre>
</td>

<td>
<pre>
{
 bands: [
    {band: 1, palette: 'red'},
    {band: 2, palette: 'green'},
    {band: 3, palette: 'blue'}
  ]
}
</pre>
</td>

</tr>
</table>

However, they are not and produce totally different thumbnails:

|A|B|
|---|---|
| ![thumbnail](https://user-images.githubusercontent.com/22067021/167905678-6fcc9dfd-6bc9-4061-a0c6-6be0f0db1d36.png) |  ![ee1bc20c-6d63-4751-be0f-8e5fa529ce66](https://user-images.githubusercontent.com/22067021/167905712-8f2515b0-34b3-4d7c-bea2-57072b3fcbc4.png) |

Is this expected? Should `"red"` be interpreted as equivalent to `['#000', '#f00']`

**Comment by banesullivan** [comment]
Notably, C (below) yields the same result as A:

```
{
 bands: [
    {band: 1, palette: '#f00'},
    {band: 2, palette: '#0f0'},
    {band: 3, palette: '#00f'}
  ]
}
```

![thumbnail](https://user-images.githubusercontent.com/22067021/167906977-10008e9d-7980-4d49-8dee-2f2332694269.png)

**Comment by manthey** [comment]
I don't think there is a good answer here -- if we muck with what `green` means compared to css it will cause surprise.  If we don't, it will cause surprise.  Maybe we should define `R`, `G`, `B` and `RED`, `GREEN`, `BLUE` to be `#f00`, `#0f0, `#00f`, since the capitalized versions aren't used.

**Comment by banesullivan** [comment]
> if we muck with what green means compared to css it will cause surprise.

Since this is an image-focused library, I think "green" has a specific meaning in this context to refer to the Green channel of an RGB image which should be mapped to `#00ff00` and not `#008000`

IMO, it is more surprising to follow a CSS convention than the convention of the RGB color model here

> Maybe we should define `R`, `G`, `B` and `RED`, `GREEN`, `BLUE` to be `#f00`, `#0f0, `#00f`, since the capitalized versions aren't used.

That's a fair compromise, but I think the following would be clunky:

```
{
 bands: [
    {band: 1, palette: 'red'},
    {band: 2, palette: 'GREEN'},
    {band: 3, palette: 'blue'}
  ]
}
```
Though the following looks nice to me and we could document that `R`, `G`, and `B` should be used when wanting to map like the channels of an RGB image
```
{
 bands: [
    {band: 1, palette: 'R'},
    {band: 2, palette: 'G'},
    {band: 3, palette: 'B'}
  ]
}
```

**Comment by banesullivan** [comment]
No matter what, this should be documented in `docs/tilesource_options.rst`

**Comment by manthey** [comment]
> Since this is an image-focused library, I think "green" has a specific meaning in this context to refer to the Green channel of an RGB image which should be mapped to `#00ff00` and not `#008000`
> 
> IMO, it is more surprising to follow a CSS convention than the convention of the RGB color model here

Except that a lot of the names are from css (e.g., `purple` is half-intensity `magenta`, `violet` seems to have little to do with the spectral color).

> > Maybe we should define `R`, `G`, `B` and `RED`, `GREEN`, `BLUE` to be `#f00`, `#0f0`, `#00f`, since the capitalized versions aren't used.
> 
> That's a fair compromise, but I think the following would be clunky:

Yes -- that's why using all caps would be mapped for red and blue, too.  `r`, `g`, `b` are synonyms to `red`, `green`, `blue`, so we can't reuse those.

**Comment by manthey** [comment]
This was resolved in #858.
